core/src/test/java/org/elasticsearch/fieldstats/FieldStatsTests.java

```diff
@@ -519,49 +600,84 @@ public void testMetaFieldsNotIndexed() {
     }
 
     public void testSerialization() throws IOException {
+        Version version = randomBoolean() ? Version.CURRENT : Version.V_5_1_0_UNRELEASED;
         for (int i = 0; i < 20; i++) {
-            assertSerialization(randomFieldStats());
+            assertSerialization(randomFieldStats(version.onOrAfter(Version.V_5_2_0_UNRELEASED)),
+                version);
         }
     }
 
     /**
      * creates a random field stats which does not guarantee that {@link FieldStats#maxValue} is greater than {@link FieldStats#minValue}
      **/
-    private FieldStats randomFieldStats() throws UnknownHostException {
+    private FieldStats randomFieldStats(boolean withNullMinMax) throws UnknownHostException {
         int type = randomInt(5);
```

Comment:
I think it's fine to use the switch statement as long as the `default` throws an exception ?